5.machine_learning/scripts/2.evalute_models.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Evaluate model performance with PR curves, confusion matrixes, and F1 scores
+
+# ## Import libraries
+
+# In[1]:
+
+
+from joblib import load
+import numpy as np
+import matplotlib.pyplot as plt
+import pathlib
+import pandas as pd
+import seaborn as sns
+from sklearn.metrics import precision_recall_curve
+
+import sys
+
+sys.path.append("../utils")
+from eval_utils import generate_confusion_matrix_df, generate_f1_score_df
+from training_utils import load_data
+
+
+# ## Set paths to different datasets and models
+
+# In[2]:
+
+
+# Directory with datasets
+data_dir = pathlib.Path("./data")
+
+# Directory with models
+models_dir = pathlib.Path("./models")
+
+# Directory for model figures output
+fig_dir = pathlib.Path("./figures")
+fig_dir.mkdir(exist_ok=True)
+
+# Load in each model individually
+final_model = load(pathlib.Path(f"{models_dir}/log_reg_fs_plate_4_final_downsample.joblib"))
+shuffled_model = load(pathlib.Path(f"{models_dir}/log_reg_fs_plate_4_shuffled_downsample.joblib"))
+
+
+# ## Precision-Recall Curves
+
+# ### Output PR data per dataset and combine into one df
+
+# In[3]:
+
+
+# Initialize empty lists to store data for each iteration
+precision_list = []
+recall_list = []
+threshold_list = []
+model_type_list = []
+data_type_list = []
+
+for model_path in models_dir.iterdir():
+    print("Evaluating", model_path.stem.split("_")[5], "model...")
```

Comment:
Understandable to try and avoid hardcoding. In this case, I could use a dictionary to assign paths to models with the model type (e.g., shuffled or final) to avoid the hardcode, but that would take some time to refactor the structure. I will keep this in mind for the future, so thanks! 😄